content/blog/zardshard/2023-08-18_[gsoc_2023]_improving_icon-o-matic_final_report.md

```diff
@@ -0,0 +1,92 @@
++++
+type = "blog"
+title = "[GSoC 2023] Improving Icon-O-Matic Final Report"
+author = "Zardshard"
+date = "2023-08-18 11:21:50-04:00"
+tags = ["haiku", "software", "icon-o-matic", "gsoc", "gsoc2023"]
++++
+
+## What is Icon-O-Matic?
+
+![Screenshot of Icon-O-Matic](/files/blog/zardshard/icon_o_matic.png)
+
+There's a good chance that not everyone reading this article will know what Icon-O-Matic is, so I'll start by explaining what it is. Icon-O-Matic is a vector graphics editing program like Illustrator or Inkscape. It is specifically made to work with Haiku's custom HVIF vector graphics format. This format is similar to the SVG format, except optimized to be much, much smaller. The blog post ["500 Byte Images: The Haiku Vector Icon Format"](http://blog.leahhanson.us/post/recursecenter2016/haiku_icons.html) provides a more in-depth discussion for those interested.
+
+If you want to try Icon-O-Matic yourself, you will need to first install Haiku OS. Afterwards, you can launch it from the applications menu.
+
+## My goals
+
+The goals of my project were rather broad: to add features, refactor code, and fix bugs. I had, therefore, a lot of leeway to do what interested me. My early plans were detailed in a blog post entitled ["[GSoC 2023] Plans for improving Icon-O-Matic"](https://www.haiku-os.org/blog/zardshard/2023-05-05_gsoc_2023_plans_for_improving_icon-o-matic/). They changed rather significantly over the course of the project.
+
+## What I got done
+
+Below is a list of the things I got done during GSOC. For those interested in seeing the full list, it can be seen [here](https://review.haiku-os.org/q/owner:0azrune6%2540zard.anonaddy.com). GSoC ended for me on August 28, so commits before that date were submitted during GSoC.
```

Comment:
for the final work submission to Google, it's better if you go over these changes and add the "gsoc" hashtag to them (Gerrit has a way to do this on the sidebar).

Then you can provide a fixed link to the final commits for GSoC and nothing else, by querying by tag in addition to your email address.

Reply:
The fact that any comment or edit changes the date is one more reason to find a more permanent way to identify the commits. I don't think the way you have done it now will satisfy Google requirements (the people running GSoC need to keep track of the work done in the long term, to justify the continuation of the program at Google and also for legal reasons since they are paying for it).

The "real" date will be preserved in the git commit itself.